libass/ass_types.h

```diff
@@ -120,8 +120,12 @@ typedef struct ass_style {
     int Alignment; // use `VALIGN_* | HALIGN_*` as value
     int MarginL;
     int MarginR;
-    int MarginV;
+    /** On the next API-break (if ever happening), MarginV will be renamed to MarginT
+        and always only refer to the top margin regardless of format version */
+    int MarginV; // in SSA v4 and ASS (v4+) for both top and bottom, in v4++ only top
+    int MarginB; //v4++: bottom; v4 and v4+: identical to MarginV
     int Encoding;
+    int RelativeTo; // currently ignored, but should not be set to any value but zero
     int treat_fontname_as_pattern; // does nothing (left in place for ABI-compatibility)
```

Comment:
Shall we drop this field while we’re at it?

You asked about it in #536: this field hasn’t done anything since the introduction of the new multi-platform font provider infrastructure in 0.13.0 in 2015 (c22a4ff9a395546637dbe0f1e9d0ee549dd0069a)—which, by the way, covers the entire lifetime of libass.so.9—and I don’t think it can ever be feasibly reintroduced even if someone were to step up and ask for it (which nobody has), as the point was to ask Fontconfig to parse the font “name” as an entire pattern (with arbitrary font properties encoded in it), but other font providers, of course, have no idea what a Fontconfig pattern even is.

I’m willing to bet that this field has never been _set_ by anyone other than MPlayer and its derivatives. mpv has patched it out in 2015, when it became obsolete. Unfortunately, MPlayer is [still setting it](https://github.com/sherpya/MPlayer/blob/4cc95fdf90021978958a36c72dd52bf97868049d/sub/ass_mp.c#L63) (I’ve just confirmed with Subversion trunk).

Reply:
The API part is precisely why I mentioned MPlayer. I think it is safe to think that only MPlayer proper would be affected—and if we could submit a patch for it, maybe this could be neglected. But maybe not (probably depends on how much we want to support older MPlayer and how soon the next MPlayer release might happen).

Reply:
> how soon the next MPlayer release might happen

MPlayer 1.5 happened to be released in February with an important point in the release notes being support for ffmpeg 5.x API (the `ass_set_storage_size` patch made it just in time). 1.4 was in April 2019 and mentioned ffmpeg 4.1 support, 1.3 in February 2016 for ffmpeg 3.0.x support. Additionally all mentioned release note recommend using svn trunk. Based on this I don't expect a new MPlayer release anytime soon. Distro's would need to backport a `treat_fontname_as_pattern` patch.